### ISSUE TITLE
关于docker容器里面无法初始化

### ISSUE THREAD

**Issue by tianpengg** [kazuyaujihara/build-rdkit#5]
您好，在docker运行时，报错，如下“The type initializer for 'GraphMolWrap.RDKFuncsPINVOKE' threw an exception.”，请问这需要配置其他的操作嘛

**Comment by k-ujihara** [comment]
Please let us know your `Dockerfile`.

**Comment by tianpengg** [comment]
> 请让我们知道您的`Dockerfile`.

Could you please provide a Dockerfile,my Dockerfile just for.Net Core,
thanks

**Comment by dmtregrv** [comment]
Some libs required for a normal working present in WSL, but missed in Docker images. I fixed it with this:
`RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -y libpixman-1-0 libpng16-16 libx11-6 libbz2-1.0 
`

UPD.: for WSL users, which haven't installed python3-rdkit packages and also have such problems (as me, on my new machine) libpixman-1-0 is the only missing part.
`sudo apt install libpixman-1-0`

**Comment by zealseeker** [comment]
Hi @kazuyaujihara 
I am using mcr.microsoft.com/dotnet/sdk:5.0. Rdkit installed via nuget does not work.

Here is the way to reproduce:

1. `docker run -it mcr.microsoft.com/dotnet/sdk:5.0 bash`
2. `dotnet new console`
3. `dotnet add package RDKit.DotNetWrap --version 0.2020094.2`
4. Install vim and edit Program.cs and add something like `"RWMol rwMol = RWMol.MolFromSmiles("c1cccc1");`
5. `dotnet build && dotnet run`

It can be built successfully, but there is error when running:
```
Unhandled exception. System.TypeInitializationException: The type initializer for 'GraphMolWrap.RDKFuncsPINVOKE' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'SWIGExceptionHelper' threw an exception.
 ---> System.DllNotFoundException: Unable to load shared library 'RDKFuncs' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libRDKFuncs: cannot open shared object file: No such file or directory
```

If I add `~/.nuget/packages/rdkit.dotnetwrap/0.2020094.2/runtimes/linux-x64/native` to `LD_LIBRARY_PATH`, there will be `Segmentation fault (core dumped)`

It seems that #3 still exists.

**Comment by k-ujihara** [comment]
I understand the problem. I used Ubuntu on WSL2 when making NuGet package and it may cause problems. I will consider creating a Dockerfile, but I welcome your help.

**Comment by zealseeker** [comment]
Hi @kazuyaujihara 
I've solved the problem. There are three points to be noted:
1. Some dependencies need to be installed as mentioned by @dmtregrv So it should be added to the docker file.
2. When compiling, we need to explicitly set the runtime (`dotnet build -r linux-x64 -c Release`)
3. LD_LIBRARY_PATH should be set to `$PWD/bin/Release/net5.0/linux-x64`

2&3 is slightly different from https://github.com/kazuyaujihara/build-rdkit/issues/6#issuecomment-974577709
where you did not assign the runtime.  In my case, if I don't set the runtime and use the LD_LIBRARY_PATH like `bin/Release/net5.0/runtime/linux-x64/native`, it will be Segmentation fault.

**Comment by sonicning** [comment]
I had the same issue，and I used the solution above, it still has Segmentation fault. After changing the base image from Debian to Ubuntu, the problem solved

**Comment by dmtregrv** [comment]
@sonicning 
Missing libs. I use Microsoft images based on Ubuntu, so I was looking for necessary libs for my case. If you want to switch distro to Debian you could find requared libs by setting 

> export LD_DEBUG=libs

**Comment by k-ujihara** [comment]
This may be fixed in this release [v0.2021094.2](https://github.com/kazuyaujihara/build-rdkit/releases/tag/v0.2021094.2). 
See  [Instruction to Build Application](https://github.com/kazuyaujihara/build-rdkit/tree/1edd9c1cf9148d8e0f147173fb71e7fdfa241e70/myApp). [Dockerfile](https://github.com/kazuyaujihara/build-rdkit/blob/1edd9c1cf9148d8e0f147173fb71e7fdfa241e70/myApp/Dockerfile) is also available.

**Comment by k-ujihara** [comment]
Fixed by https://github.com/kazuyaujihara/build-rdkit/commit/43734a374102c7017b337fc7e30be1dbddede459.
